pkg/api/api.go

```diff
@@ -10,13 +10,34 @@ type ErrorResp struct {
 // BlockVolumeCreateReq represents gluster block
 // volume create request
 type BlockVolumeCreateReq struct {
-	HaCount            int      `json:"ha"`
-	AuthEnabled        bool     `json:"auth-enabled"`
-	FullPrealloc       bool     `json:"full-prealloc"`
-	Size               uint64   `json:"size"`
-	Storage            string   `json:"storage"`
-	RingBufferSizeInMB uint64   `json:"ring-buffer-size-mb"`
-	Hosts              []string `json:"hosts"`
+	// Name represents block Volume name
+	Name               string      `json:"name,omitempty"`
+	HostingVolume      string      `json:"hosting_volume,omitempty"`
+	HaCount            int         `json:"ha,omitempty"`
+	AuthEnabled        bool        `json:"auth-enabled"`
+	FullPrealloc       bool        `json:"full-prealloc"`
+	Size               interface{} `json:"size,omitempty"`
+	Storage            string      `json:"storage,omitempty"`
+	RingBufferSizeInMB uint64      `json:"ring-buffer-size-mb,omitempty"`
+	Hosts              []string    `json:"hosts,omitempty"`
```

Comment:
Hosts is not a required param for block volume creation?

Reply:
Hosts are used in gluster-block cli

Reply:
why its omitempty?